src/server/pps/cmds/cmds.go

```diff
@@ -619,9 +615,169 @@ each datum.`,
 			return nil
 		}),
 	}
-	queryLoki.Flags().StringVar(&since, "since", "0", "Return log messages more recent than \"since\".")
+	queryLoki.Flags().DurationVar(&since, "since", 0, "Return log messages more recent than \"since\".")
 	commands = append(commands, cmdutil.CreateAlias(queryLoki, "loki"))
 
+	tailLoki := &cobra.Command{
+		Use:   "{{alias}}",
+		Short: "Tail a loki query.",
+		Long:  "Tail a loki query.",
+		Run: cmdutil.RunFixedArgs(0, func(s []string) error {
+			ctx, done := pctx.Interactive()
+			defer done()
+
+			sleep := func() {
+				select {
+				case <-ctx.Done():
+				case <-time.After(time.Second):
+				}
+			}
+
+			var connected bool
+		main:
+			for {
+				// Attempt to stay connected forever, so that tailing can continue
+				// when pachd restarts.
+				select {
+				case <-ctx.Done():
+					return nil
+				default:
+				}
+
+				client, err := pachctlCfg.NewOnUserMachine(ctx, false)
+				if err != nil {
+					if !connected {
+						// If the first connection fails, it's probably
+						// never going to work.  Give up.
+						return errors.Wrap(err, "dial")
+					}
+					log.Error(ctx, "attempt to reconnect to server failed; retrying", zap.Error(err))
+					sleep()
+					continue
+				}
+				c, err := client.TailLoki(client.Ctx(), &pps.LokiRequest{
+					Since: types.DurationProto(since),
+					Query: lokiQuery,
+				})
+				if err != nil {
+					if !connected {
+						return errors.Wrap(err, "start tail")
+					}
+					log.Error(ctx, "attempt to start tail stream failed; retrying", zap.Error(err))
+					sleep()
+					continue
+				}
+				for {
+					msg, err := c.Recv()
+					if err != nil {
+						if errors.Is(err, io.EOF) {
+							time.Sleep(time.Second)
+							continue main
+						}
+						if !connected {
+							return errors.Wrap(err, "recv")
+						}
+						log.Error(ctx, "failed to recv message; retrying", zap.Error(err))
+						sleep()
+						continue main
+					}
+					connected = true
+					switch x := msg.GetResponse().(type) {
+					case *pps.TailLokiResponse_Warning:
+						log.Info(ctx, "server warning: "+x.Warning)
```

Comment:
How useful is it for the user to be alerted here? Is it worth making this log.Error to be more visible to the user?